crates/oxc_linter/src/rules/eslint/id_length.rs

```diff
@@ -48,24 +50,41 @@ impl Deref for IdLength {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase", default)]
 pub struct IdLengthConfig {
+    /// An array of regex patterns for identifiers to exclude from the rule.
+    /// For example, `["^x.*"]` would exclude all identifiers starting with "x".
+    #[schemars(with = "Vec<String>")]
     exception_patterns: Vec<Regex>,
+    /// An array of identifier names that are excluded from the rule.
+    /// For example, `["x", "y", "z"]` would allow single-letter identifiers "x", "y", and "z".
     exceptions: Vec<String>,
+    /// The maximum number of graphemes allowed in an identifier.
+    /// Defaults to no maximum (effectively unlimited).
     max: u64,
+    /// The minimum number of graphemes required in an identifier.
     min: u64,
+    /// When set to `"never"`, property names are not checked for length.
+    /// When set to `"always"` (default), property names are checked just like other identifiers.
     properties: PropertyKind,
 }
 
-impl Default for IdLength {
+impl Default for IdLengthConfig {
     fn default() -> Self {
-        Self(Box::new(IdLengthConfig {
```

Comment:
@camc314 This change may not be safe? I can undo it if we'd rather not mess with it, but the tests do pass and it seems to work fine.